install/tools/istio_vm_common.sh

```diff
@@ -0,0 +1,216 @@
+function istioGenerateClusterConfigs() {
+   local K8SCLUSTER=${1}
+
+   local NS=${ISTIO_NAMESPACE:-istio-system}
+
+  # Multiple tries, it may take some time until the controllers generate the IPs
+  for i in {1..10}
+  do
+    PILOT_IP=$(kubectl get -n $NS service istio-pilot-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    ISTIO_DNS=$(kubectl get -n kube-system service dns-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    MIXER_IP=$(kubectl get -n $NS service mixer-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    CA_IP=$(kubectl get -n $NS service istio-ca-ilb -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+    if [ "${PILOT_IP}" == "" -o  "${ISTIO_DNS}" == "" -o "${MIXER_IP}" == "" ] ; then
+        echo Waiting for ILBs
+        sleep 5
+    else
+        break
+    fi
+  done
+
+  if [ "${PILOT_IP}" == "" -o  "${ISTIO_DNS}" == "" -o "${MIXER_IP}" == "" ] ; then
+    echo "Failed to create ILBs"
+    exit 1
+  fi
+
+  #/etc/dnsmasq.d/kubedns
+  echo "server=/svc.cluster.local/$ISTIO_DNS" > kubedns
+  echo "address=/istio-mixer/$MIXER_IP" >> kubedns
+  echo "address=/mixer-server/$MIXER_IP" >> kubedns
+  echo "address=/istio-pilot/$PILOT_IP" >> kubedns
+  echo "address=/istio-ca/$CA_IP" >> kubedns
+
+  CIDR=$(gcloud container clusters describe ${K8SCLUSTER} $(_istioGcloudOpt) --format "value(servicesIpv4Cidr)")
+  echo "ISTIO_SERVICE_CIDR=$CIDR" > cluster.env
+}
+
+
+# Install required files on a VM and run the setup script.
+#
+# Must be run for each VM added to the cluster
+# Params:
+# - name of the VM - used to copy files over.
+# - service account to be provisioned (defaults to istio.default)
+function istioProvisionVM() {
+ NAME=${1}
+
+ local SA=${2:-istio.default}
+
+  kubectl get secret $SA -o jsonpath='{.data.cert-chain\.pem}' |base64 -d  > cert-chain.pem
+  kubectl get secret $SA -o jsonpath='{.data.root-cert\.pem}' |base64 -d  > root-cert.pem
+  kubectl get secret $SA -o jsonpath='{.data.key\.pem}' |base64 -d  > key.pem
+
+  local ISTIO_FILES=${ISTIO_FILES:-.}
+
+ # Copy deb, helper and config files
+ # Reviews not copied - VMs don't support labels yet.
+ istioCopy $NAME \
+   kubedns \
+   *.pem \
+   cluster.env \
+   $ISTIO_FILES/istio_vm_setup.sh \
+   $ISTIO_FILES/istio-proxy-envoy.deb \
+   $ISTIO_FILES/istio-agent.deb \
+   $ISTIO_FILES/istio-auth-node-agent.deb \
+
+ # Run the setup script.
+ istioRun $NAME "sudo bash -c -x ./istio_vm_setup.sh"
+}
+
+
+# Helper functions for the main script
+
+# If Istio was built from source, copy the artifcats to the current directory, for use
+# by istioProvisionVM
+function istioCopyBuildFiles() {
+  local ISTIO_IO=${ISTIO_BASE:-${GOPATH:-$HOME/go}}/src/istio.io
+  local ISTIO_FILES=${ISTIO_FILES:-.}
+
+  (cd $ISTIO_IO/proxy; bazel build tools/deb/... )
+  (cd $ISTIO_IO/pilot; bazel build tools/deb/... )
+  (cd $ISTIO_IO/auth; bazel build tools/deb/... )
+
+  cp $ISTIO_IO/proxy/bazel-bin/tools/deb/istio-proxy-envoy.deb \
```

Comment:
Are these .deb files only working on certain platforms?

Reply:
I tested on current debian and xenial. Should work on other similar platforms.

Reply:
Only tested on current deb and xenial. 
For Trusty would be tricky ( envoy c++ deps ), but it is possible with a special build.